compiler/options.nim

```diff
@@ -587,6 +588,17 @@ proc pathSubs*(conf: ConfigRef; p, config: string): string =
   if "~/" in result:
     result = result.replace("~/", home & '/')
 
+proc nimbleSubs*(conf: ConfigRef; p: string): seq[string] =
+  let pl = p.toLowerAscii
+  if "$nimblepath" in pl or "$nimbledir" in pl:
+    for nimblePath in conf.nimblePaths:
+      let nimblePath = removeTrailingDirSep(nimblePath.string)
+      let p = p % ["nimblepath", nimblePath, "nimbledir", nimblePath]
+      if p notin result:
+        result.add p
+  else:
+    result.add p
+
```

Comment:
Should we not simply generalize `pathSubs` to be an iterator that can yield 1+ values?

Reply:
Would cause too much impact - pathSubs is used in other places too.

Reply:
Right, that's the idea.

Reply:
`pathSubs()` is also used by the `patchFile()` nimscript proc as well as `getModuleName()` calls. I don't think either benefits from a `$nimbleDir` substitution, Araq might know better. Even if we change `pathSubs()` to return a `seq[AbsolutePath]` and pick `[0]`, it will propagate all over the place since `processPath()` would also need to return a seq and is called in many places which are irrelevant for `$nimbleDir`.

Alternative is to simply substitute `$nimbleDir` with the last nimbleDir added to Nim in `pathSubs()` and call it a day but the current implementation is marginally better functionality without adding too much code with no benefit.

Reply:
It's not about what benefits from it; we cannot know the use of this code, so it should be made as consistent as possible.  If it "propogates all over the place", so much the better; that is in fact what we want!

A `pathSubs` iterator won't clash with a proc, but you can rename it if you want -- I just want a way to automatically use a `$nimbledir` substitution when one exists, as I currently do for `pathSubs`.

This is not "no benefit"; it's solving a real need for the tools that use these substitutions.